### PR TITLE
ViscoPlastic rheology migrated into separate rheology module

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR splits the rheological parts of the ViscoPlastic material model into a new Rheology model.
The intention is to make it easier for users to use the viscoplastic rheology in their own material models while using a different equation of state. 

* [x] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).
* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: include/aspect/material_model/visco_plastic.h
Comment (naliboff) [inline include/aspect/material_model/visco_plastic.h:239]: I think the documentation for this object should be expanded quite a bit. While one the functions does compute the viscosity, this object also points to functions for additional material model outputs, the Newton derivatives, etc. Perhaps a more accurate single sentence description would be 'Object for rheology operations'. However, my preference would be to have an expanded description.

**Review by naliboff — CHANGES_REQUESTED** [review]
👏 👏 👏 👏  - A round of applause, this is fantastic! I have one comment regarding documentation, but that is it for now. 

I think one of the other developers should have a look as well before merging, but in the end this is just moving code and I'm having a hard time anticipating any issues that the tester would not catch. @anne-glerum @MFraters @jdannberg @gassmoeller - Does one of you have time to take a look over the next few days?

**Comment by bobmyhill** [comment]
Thanks for the quick review! I've extended the documentation as requested.

**Review by naliboff — APPROVED** [review]
@bobmyhill - Thanks, the new documentation looks great! Good to go from my end, but lets wait for another review.

**Review by anne-glerum — APPROVED** [review]
Thanks @bobmyhill ! I've only got textual/formatting comments, and most of those are about things not introduced by this PR. Not sure whether those should be addressed in a separate PR or can just be added to this PR. What do you think @naliboff ?

**Comment by naliboff** [comment]
@anne-glerum - Thanks for the second review! If @bobmyhill does not mind documenting the parameters mentioned above, maybe easiest to do it in this PR? Bob - if you are ok with this, I'll add some suggestions for the missing parameter documentation to speed things up.

**Comment by bobmyhill** [comment]
Thanks @anne-glerum! The commenting is much more uniform now :) 
@naliboff, I've made the changes suggested. Is anything else needed, or is this ready to go now (once the tester confirms that it is still happy)?

File: include/aspect/material_model/rheology/visco_plastic.h
Comment (naliboff) [inline include/aspect/material_model/rheology/visco_plastic.h:196]: Suggestion: `Reference strain rate for the first non-linear iteration in the first time step`

File: include/aspect/material_model/rheology/visco_plastic.h
Comment (naliboff) [inline include/aspect/material_model/rheology/visco_plastic.h:168]: Should this be` strain rate dependent rheology`?

**Review by naliboff — COMMENTED** [review]
@bobmyhill - Two more minor comments on my end (feel free to ignore them), but after that ready to merge. Thank you again for this major improvement!

**Comment by bobmyhill** [comment]
Ok, great! I've made those changes, so this is now ready to go :)

**Pull request merged**
